velox/exec/SortWindowBuild.cpp

```diff
@@ -179,15 +200,20 @@ void SortWindowBuild::computePartitionStartRows() {
   partitionStartRows_.push_back(0);
 
   VELOX_CHECK_GT(sortedRows_.size(), 0);
-  for (auto i = 1; i < sortedRows_.size(); i++) {
-    if (partitionCompare(sortedRows_[i - 1], sortedRows_[i])) {
-      partitionStartRows_.push_back(i);
-    }
-  }
 
-  // Setting the startRow of the (last + 1) partition to be returningRows.size()
-  // to help for last partition related calculations.
-  partitionStartRows_.push_back(sortedRows_.size());
+  // Currently, computePartitionStartRows find the partition start rows is a
+  // linear search algorithm, the time complexity is O(n). However, for many
+  // scenarios where the data partitions are the same, the search algorithm is
+  // inefficient. It can be optimized to binary search, but the time complexity
+  // of back-to-back binary search will become worse when there are many
+  // partitions. Using the forward-to-back double search algorithm, will have
+  // better results when there are many or few partitions.
+  vector_size_t start = 0;
+  while (start < sortedRows_.size()) {
+    auto next = findNextPartitionStartRow(start, partitionCompare);
```

Comment:
partitionCompare function is not used in the parent function anymore. So you can define it within findNextPartitionStartRow itself instead of passing it as a parameter.

Reply:
Done.